File: ApplicationForFixingWashingMachines/src/main/java/com/example/applicationforfixingwashingmachines/entity/Problem.java
package com.example.applicationforfixingwashingmachines.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.Where;

import java.time.LocalDateTime;
import java.util.LinkedList;
import java.util.List;

@Entity
@Getter
@Setter
@ToString
@SQLDelete(sql = "UPDATE burmistrov.public.problems SET deleted = true WHERE id=?")
@Where(clause = "deleted=false")
@Table(name = "problems")
public class Problem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String description;

    private double price;

    private LocalDateTime finishDate;

    private boolean workersFound = Boolean.FALSE;

    private boolean deleted = Boolean.FALSE;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "washingMachines_id")
    private WashingMachine washingMachine;

    @OneToMany( mappedBy = "problem", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @ToString.Exclude
    private List<Repair> repairs = new LinkedList<>();

}

File: frontend/src/app/models/Problem.ts
import {Repair} from "./Repair";

export interface Problem {
  id: number;
  description: string;
  price: number;
  finishDate:Date;
  washingMachineId: number;
  workersFound: boolean;
  repairs: Array<Repair>;
}

export class ProblemImpl implements Problem{
  description: string = "";
  finishDate: Date = new Date();
  id: number = -1;
  price: number = 0;
  repairs: Array<Repair> = [];
  workersFound = false;
  washingMachineId: number = -1;

}
